### PR TITLE
Fix scalac warnings up to 2.12

### PR DESCRIPTION
**Pull request by htmldoug**
Turns on `"-Xfatal-warnings"` at scala 2.12 and fixes errors.

@russellremple

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -11,7 +11,12 @@ import com.typesafe.tools.mima.core._
 val scalaVersions = Seq("2.11.12", "2.12.8", "2.13.0")
 
 trait CommonModule extends ScalaModule {
-  def scalacOptions = T{ if (scalaVersion() startsWith "2.12") Seq("-opt:l:method") else Nil }
+
+  def scalacOptions = T {
+    // Not ready to deal with 2.13 collection deprecations.
+    (if (scalaVersion() startsWith "2.12") Seq("-opt:l:method", "-Xfatal-warnings", "-deprecation", "-feature") else Nil)
```
Comment (jducoeur) [inline build.sc]: You might want to think about enabling `higherKinds` at this level -- it's on the tpolecat list, which I tend to take as the benchmark for appropriate flags to use, and would save you from having to add it file-by-file.  (Frankly, I think it's a bit goofy that this one requires a flag.)

File: weejson/src/com/rallyhealth/weejson/v1/Value.scala [hunk 2]
```diff
@@ -209,8 +207,8 @@ object Obj{
   // type inference issues due to overloading the existing `apply` method
   // generated by the case class itself
   // https://github.com/lihaoyi/upickle/issues/230
-  def apply[V <% Value](item: (String, V),
-                        items: (String, Value)*): Obj = {
+  def apply[V](item: (String, V),
+                        items: (String, Value)*)(implicit viewBound: V => Value): Obj = {
```
Comment (jducoeur) [inline weejson/src/com/rallyhealth/weejson/v1/Value.scala]: Wow, haven't seen a view bound in the wild in a while...

File: build.sc (hunk 1 above)
Comment (htmldoug) [reply build.sc]: You're the second person to tell me to just enable it at this level. I guess I'll do that!

File: weejson/src/com/rallyhealth/weejson/v1/Value.scala (hunk 2 above)
Comment (htmldoug) [reply weejson/src/com/rallyhealth/weejson/v1/Value.scala]: rite?

**Comment by russellremple** [comment]
The name viewBound seems kind of generic. Maybe vAsValue or something like
that?

**Comment by htmldoug** [comment]
They both sound equally generic to me.

**Pull request merged**